File: lua/perfect-numbers/perfect-numbers.lua
local function aliquot_sum(n)
  local sum = 0;
  for f = 1, n - 1 do
    if (n % f == 0) then
      sum = sum + f
    end
  end
  return sum;
end

local function classify(n)
  local sum = aliquot_sum(n)
  return n == sum and 'perfect' or n < sum and 'abundant' or 'deficient'
end

return { aliquot_sum = aliquot_sum, classify = classify }

File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    """A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if number <= 0:
        raise ValueError("Classification is only possible for positive integers.")

    s = sum(f if number % f == 0 else 0 for f in range(1, number))

    return "perfect" if s == number else "abundant" if number < s else "deficient"
